custom_components/hyxi_cloud/sensor.py

```
@@ -1459,15 +1459,25 @@ def _update_native_value(self):
         if key == "acE" and (value is None or str(value) == "0.0"):
             if getattr(self, "_device_type", None) in (
                 "grid_connected_inverter",
                 "micro_inverter",
             ):
                 value = metrics.get("efpv")
 
+        if key == "gridF" and value is None:
+            if getattr(self, "_device_type", None) in (
+                "grid_connected_inverter",
+                "micro_inverter",
+            ):
```

Ruff (0.15.20)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)